ruby/src/api.cc

```diff
@@ -0,0 +1,509 @@
+#include <leatherman/ruby/api.hpp>
+#include <leatherman/util/environment.hpp>
+#include <leatherman/execution/execution.hpp>
+#include <leatherman/logging/logging.hpp>
+#include <boost/algorithm/string.hpp>
+#include <boost/filesystem/path.hpp>
+#include <boost/filesystem/operations.hpp>
+#include <sstream>
+
+using namespace std;
+using namespace leatherman::util;
+using namespace leatherman::execution;
+using namespace boost::filesystem;
+
+namespace lth_lib = leatherman::dynamic_library;
```

Comment:
If I try to use a `using namespace leatherman::dynamic_library` here, it complains every time `dynamic_library` appears as a type, with this: `error: must use 'struct' tag to refer to type 'dynamic_library' in this scope`. It's as if it's getting confused and thinks I'm referring to the namespace in places I shouldn't by. Why is this? I assume it has something to do with the struct and the namespace having the same name... is this inadvisable?